client/src/mdnplusplus/landing-page-survey/index.tsx

```diff
@@ -0,0 +1,274 @@
+import React from "react";
+import useSWR from "swr";
+
+import { useUserData } from "../../user-context";
+import "./index.scss";
+
+const API_URL = "/api/v1/mdnplusplus/landing-page-survey/";
+const SESSIONSTORAGE_KEY_UUID = "mdnplusplus-landing-page-survey-uuid";
+const SESSIONSTORAGE_KEY_EMAIL = "mdnplusplus-landing-page-survey-email";
+
+interface PingData {
+  csrfmiddlewaretoken: string;
+  uuid: string;
+}
+
+function setSessionStorageData(key: string, value: string) {
+  try {
+    sessionStorage.setItem(key, value);
+  } catch (err) {
+    console.warn("Unable to set sessionStorage key");
+  }
+}
+
+function getSessionStorageData(key: string) {
+  try {
+    return sessionStorage.getItem(key);
+  } catch (err) {
+    console.warn("Unable to set sessionStorage key");
+  }
+}
+
+export default function LandingPageSurvey({ variant }: { variant: number }) {
+  const [email, setEmail] = React.useState(
+    getSessionStorageData(SESSIONSTORAGE_KEY_EMAIL) || ""
+  );
+  const [page, setPage] = React.useState<"start" | "thankyou" | "price">(
+    "start"
+  );
+  const userData = useUserData();
+  React.useEffect(() => {
+    if (userData && userData.email && !email) {
+      setEmail(userData.email);
+    }
+  }, [userData, email]);
+
+  React.useEffect(() => {
+    if (email) {
+      setSessionStorageData(SESSIONSTORAGE_KEY_EMAIL, email.trim());
+    }
+  }, [email]);
+
+  const [price, setPrice] = React.useState("");
+
+  const [
+    surveySubmissionError,
+    setSurveySubmissionError,
+  ] = React.useState<Error | null>(null);
+
+  // Use a useMemo(() => {...}, [variant]) so that you don't get a different
+  // response if there's a re-render because in a re-render you might get
+  // something different (from the first render) from getSessionStorageData().
+  const pingURL = React.useMemo(() => {
+    const pingSP = new URLSearchParams();
+    pingSP.set("variant", `${variant}`);
+    const previousUUID = getSessionStorageData(SESSIONSTORAGE_KEY_UUID);
+    if (previousUUID) {
+      pingSP.set("uuid", previousUUID);
+    }
+    return `${API_URL}?${pingSP.toString()}`;
+  }, [variant]);
+
+  const { data: pingData, error: pingError } = useSWR<PingData>(
+    pingURL,
+    async (url) => {
+      const response = await fetch(url);
+      if (!response.ok) {
+        throw new Error(`${response.status} on ${url}`);
+      }
+      return await response.json();
+    },
+    {
+      revalidateOnFocus: false,
+    }
+  );
+
+  React.useEffect(() => {
+    if (pingData) {
+      setSessionStorageData(SESSIONSTORAGE_KEY_UUID, pingData.uuid);
+    }
+  }, [pingData]);
+
+  async function sendSurveySubmission() {
+    if (!pingData) {
+      throw new Error("Can't send survey if ping didn't work");
+    }
+    const formData = new URLSearchParams();
+    formData.set("response", JSON.stringify({ price }));
+    formData.set("variant", `${variant}`);
```

Comment:
Nit. This is not necessary, since the `variant` has already been saved in the record uniquely identified by the `uuid`, and also because it's ignored by the backend code when POSTing to the `API_URL` anyway.